src/vm/methodtablebuilder.cpp

```diff
@@ -9519,42 +9519,115 @@ void MethodTableBuilder::CheckForSystemTypes()
 {
     STANDARD_VM_CONTRACT;
 
+    LPCUTF8 name, nameSpace;
+
     MethodTable * pMT = GetHalfBakedMethodTable();
     EEClass * pClass = GetHalfBakedClass();
 
     // We can exit early for generic types - there are just a few cases to check for.
-    if (bmtGenerics->HasInstantiation() && g_pNullableClass != NULL)
+    if (bmtGenerics->HasInstantiation() && pClass->HasLayout())
```

Comment:
We will be going down the more expensive path for all generics without HasLayout now. Is it expected?

Reply:
Probably not, I'll move this to a separate `if` check.

Reply:
Fixed.

If the class doesn't have a layout, it will go down the same path as before.
If it does, it will go down the same path as before if it isn't in the `System.Runtime.Intrinsics` namespace.
If it is in the `System.Runtime.Intrinsics` namespace, we expect it to be one of the special SIMD types or explicitly skipped in the assert (which we are doing for the `DebugView` helper classes).